### PR TITLE
DBZ-6047 Upgrade to Kafka 3.5

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-6047

Related PR - https://github.com/debezium/debezium-connector-vitess/pull/143

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java [hunk 1]
```diff
@@ -153,6 +153,7 @@ public SnapshotResult<O> doExecute(ChangeEventSourceContext context, O previousO
 
             if (snapshottingTask.snapshotData()) {
                 LOGGER.info("Snapshot step 7 - Snapshotting data");
+                Thread.sleep(2000);
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java:156]: Gasp, is there any way to not do that?

Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java:156]: @gunnarmorling Sorry for that, it is garbage from a testing of a different PR :-).
I going to remove it right away.
Still please read https://issues.redhat.com/browse/DBZ-6047?focusedId=22479922&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-22479922

Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java]: @gunnarmorling Is it better now? :-)

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java]: Yepp, much better, thanks :)

File: pom.xml [hunk 2]
```diff
@@ -93,13 +93,13 @@
         <release.endpoint>https://s01.oss.sonatype.org/</release.endpoint>
 
         <!-- Kafka and it's dependencies MUST reflect what the Kafka version uses -->
-        <version.kafka>3.4.0</version.kafka>
-        <version.zookeeper>3.6.3</version.zookeeper>
+        <version.kafka>3.5.0</version.kafka>
+        <version.zookeeper>3.6.4</version.zookeeper>
```
Comment (gunnarmorling) [inline pom.xml:97]: As I'm seeing this, are there plans yet for phasing out the ZK dependency altogether? I.e. only use and show KRaft mode in embedded cluster, docs, tutorial, etc?

**Pull request merged**